moodler/students.py

```diff
@@ -21,31 +23,41 @@ def core_enrol_get_enrolled_users(course_id):
     return response
 
 
-def is_student_in_names(students_names, enrolled):
+def is_student_in_names(students_names_dict, enrolled):
     """
     Check whether the student enrolled is in the list of names received.
     """
-    for student_name in students_names:
-        if student_name in enrolled['fullname']:
+    for student_name, was_student_found in students_names_dict.items():
+        if student_name.lower() in enrolled['fullname'].lower():
             logger.info("Found student '%s' for the received name '%s'",
                         enrolled['fullname'],
                         student_name)
-            return True
+
+            if was_student_found:
+                logger.error("Already found student - skipping this name")
+                return False
+
+            students_names_dict[student_name] = True
 
     return False
 
 
-def get_students(course_id, students_names=None):
+def get_students(course_id, students_names: list = None):
```

Comment:
I'm having trouble understanding this function in this context...
Document the parameters please. What is student_names? When would this be None?
Also, students_names_dict is basically a dict of student_name to found_status? If yes, change the name to be more descriptive, maybe add a comment. students_names_dict is not informative enough...

Reply:
You are totally right. I have decided that I was trying to enforce two functions to be one function.
I created a whole new function to be used for retrieving students IDs by their names.